clang/lib/3C/3C.cpp

```diff
@@ -351,18 +351,25 @@ bool _3CInterface::addVariables() {
 
   std::lock_guard<std::mutex> Lock(InterfaceMutex);
 
+  // first step, so load the ASTs
   ClangTool &Tool = getGlobalClangTool();
+  Tool.buildASTs(ASTs);
 
-  // 1a. Add Variables.
-  std::unique_ptr<ToolAction> AdderTool = newFrontendActionFactoryA<
-      GenericAction<VariableAdderConsumer, ProgramInfo>>(GlobalProgramInfo);
+  VariableAdderConsumer VA = VariableAdderConsumer(GlobalProgramInfo, nullptr);
+  for (auto &TU : ASTs) {
+    VA.HandleTranslationUnit(TU->getASTContext());
```

Comment:
No return code form this call? `Tool.run()` below returns an exit code that could cause a return value of `false`.

Reply:
(Same question for the passes below.)

Reply:
yep. fixed.